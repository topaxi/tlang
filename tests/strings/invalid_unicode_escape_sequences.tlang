// Test for invalid Unicode escape sequences that should generate warnings
// This should compile but generate semantic warnings

fn test_invalid_unicode_escapes() {
    // These should generate warnings for invalid Unicode escape sequences
    let unterminated = "Bad escape: \u{41";         // Missing closing brace
    let empty = "Bad escape: \u{}";                 // Empty escape sequence
    let invalid_hex = "Bad escape: \u{XYZ}";        // Invalid hex characters
    let too_long = "Bad escape: \u{1234567}";       // Too many hex digits
    let invalid_codepoint = "Bad escape: \u{110000}"; // Beyond Unicode range
    let surrogate = "Bad escape: \u{D800}";         // Surrogate code point
    
    // \u not followed by { should be treated as unknown escape
    let not_unicode = "Bad escape: \uTest";
    
    // Mix valid and invalid Unicode escapes
    let mixed = "Good: \u{41}, Bad: \u{XYZ}, Good: \u{42}";
    
    // Invalid Unicode char literal
    let bad_unicode_char = '\u{110000}';
    
    log("Testing invalid Unicode escapes");
    unterminated |> log();
    empty |> log();
    invalid_hex |> log();
    too_long |> log();
    invalid_codepoint |> log();
    surrogate |> log();
    not_unicode |> log();
    mixed |> log();
    bad_unicode_char |> log();
}

test_invalid_unicode_escapes();