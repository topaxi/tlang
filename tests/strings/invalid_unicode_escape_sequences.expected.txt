Testing invalid Unicode escapes
Bad escape: \u{
Bad escape: \u{}}
Bad escape: \u{XYZ}}
Bad escape: \u{7}}
Bad escape: \u{}}
Bad escape: \u{}}
Bad escape: \uTest
Good: A, Bad: \u{XYZ}}, Good: B
\u{}}
